tests/test_graphnet_backend.py

```diff
@@ -0,0 +1,423 @@
+        ),
+        PracticeRegistrations(Patient_ID=1, StartDate="2013-01-01"),
+        backend="graphnet",
+    )
+
+    class Cohort:
+        _registrations = table("practice_registrations").first_by("patient_id")
+        arrived = _registrations.get("date_start")
+        left = _registrations.get("date_end")
+
+    assert extract(Cohort, GraphnetBackend, database) == [
+        dict(patient_id=1, arrived=datetime(2001, 1, 1), left=datetime(2012, 12, 12))
+    ]
+
+
+@pytest.mark.integration
+def test_covid_test_positive_result(database, setup_backend_database):
+    setup_backend_database(
+        Patients(Patient_ID=1),
+        PracticeRegistrations(
+            Patient_ID=1, StartDate="2001-01-01", EndDate="2026-06-26"
+        ),
+        CovidTestResults(
+            Patient_ID=1,
+            #  Remove the next line - not relevant to graphnet schema
+            # Organism_Species_Name="SARS-CoV-2",
+            SpecimenDate="2020-05-05",
+            positive_result=True,
+        ),
+        backend="graphnet",
+    )
+
+    class Cohort:
+        date = (
+            table("covid_test_results")
+            .filter(positive_result=True)
+            .earliest()
+            .get("date")
+        )
+
+    assert extract(Cohort, GraphnetBackend, database) == [
+        dict(patient_id=1, date=date(2020, 5, 5))
+    ]
+
+
+@pytest.mark.integration
+def test_covid_test_negative_result(database, setup_backend_database):
+    setup_backend_database(
+        Patients(Patient_ID=1),
+        PracticeRegistrations(
+            Patient_ID=1, StartDate="2001-01-01", EndDate="2026-06-26"
+        ),
+        CovidTestResults(
+            Patient_ID=1,
+            #  Remove the next line - not relevant to graphnet schema
+            # Organism_Species_Name="SARS-CoV-2",
+            SpecimenDate="2020-05-05",
+            positive_result=False,
+        ),
+        backend="graphnet",
+    )
+
+    class Cohort:
+        date = (
+            table("covid_test_results")
+            .filter(positive_result=False)
+            .earliest()
+            .get("date")
+        )
+
+    assert extract(Cohort, GraphnetBackend, database) == [
+        dict(patient_id=1, date=date(2020, 5, 5))
+    ]
+
+
+@pytest.mark.integration
+def test_patients_table(database, setup_backend_database):
+    setup_backend_database(
+        Patients(Patient_ID=1, Sex="F", DateOfBirth="1950-01-01"),
+        PracticeRegistrations(
+            Patient_ID=1, StartDate="2001-01-01", EndDate="2026-06-26"
+        ),
+        backend="graphnet",
+    )
+
+    class Cohort:
+        _patients = table("patients").first_by("patient_id")
+        sex = _patients.get("sex")
+        dob = _patients.get("date_of_birth")
+
+    assert extract(Cohort, GraphnetBackend, database) == [
+        dict(patient_id=1, sex="F", dob=date(1950, 1, 1))
+    ]
+
+
+""" Removing hospitalization tests - code parsing doesn't occur in the graphnet backend, so these won't work
```

Comment:
it's probably worth having one test that tests the graphnet hospitalization tests, although it won't need to check code parsing like these ones do

Reply:
Added test_hospitalization_table_returns_admission_date_and_code